libraries/AP_NavEKF2/AP_NavEKF2_PosVelFusion.cpp

```diff
@@ -214,21 +214,22 @@ bool NavEKF2_core::resetHeightDatum(void)
     // reset the height state
     stateStruct.position.z = 0.0f;
     // adjust the height of the EKF origin so that the origin plus baro height before and after the reset is the same
-    if (validOrigin) {
-        const AP_GPS &gps = AP::gps();
-        if (gps.status() >= AP_GPS::GPS_OK_FIX_3D) {
+
+    if (validOrigin && !inFlight) {
```

Comment:
We likely shouldn't allow the reset to happen at all while in flight, but since we do, not changing the origin at all seems like a bad idea.